tests/src/JIT/HardwareIntrinsics/X86/Shared/GenerateTests.csx

```diff
@@ -21,233 +21,233 @@ using System.IO;
+    ("SimpleBinOpTest.template",  new Dictionary<string, string> { ["Isa"] = "Sse", ["LoadIsa"] = "Sse", ["Method"] = "CompareOrderedScalar",                    ["RetVectorType"] = "Vector128", ["RetBaseType"] = "Single",  ["Op1VectorType"] = "Vector128", ["Op1BaseType"] =  "Single", ["Op2VectorType"] = "Vector128", ["Op2BaseType"] = "Single", ["LargestVectorSize"] = "16", ["NextValueOp1"] = "TestLibrary.Generator.GetSingle()", ["NextValueOp2"] = "TestLibrary.Generator.GetSingle()", ["ValidateFirstResult"] = "BitConverter.SingleToInt32Bits(result[0]) != ((!float.IsNaN(left[0]) && !float.IsNaN(right[0])) ? -1 : 0)",                          ["ValidateRemainingResults"] = "BitConverter.SingleToInt32Bits(left[i]) != BitConverter.SingleToInt32Bits(result[i])"}),
+    ("SimpleBinOpTest.template",  new Dictionary<string, string> { ["Isa"] = "Sse", ["LoadIsa"] = "Sse", ["Method"] = "CompareUnordered",                        ["RetVectorType"] = "Vector128", ["RetBaseType"] = "Single",  ["Op1VectorType"] = "Vector128", ["Op1BaseType"] =  "Single", ["Op2VectorType"] = "Vector128", ["Op2BaseType"] = "Single", ["LargestVectorSize"] = "16", ["NextValueOp1"] = "TestLibrary.Generator.GetSingle()", ["NextValueOp2"] = "TestLibrary.Generator.GetSingle()", ["ValidateFirstResult"] = "BitConverter.SingleToInt32Bits(result[0]) != ((float.IsNaN(left[0]) || float.IsNaN(right[0])) ? -1 : 0)",                            ["ValidateRemainingResults"] = "BitConverter.SingleToInt32Bits(result[i]) != ((float.IsNaN(left[i]) || float.IsNaN(right[i])) ? -1 : 0)"}),
+    ("SimpleBinOpTest.template",  new Dictionary<string, string> { ["Isa"] = "Sse", ["LoadIsa"] = "Sse", ["Method"] = "CompareUnorderedScalar",                  ["RetVectorType"] = "Vector128", ["RetBaseType"] = "Single",  ["Op1VectorType"] = "Vector128", ["Op1BaseType"] =  "Single", ["Op2VectorType"] = "Vector128", ["Op2BaseType"] = "Single", ["LargestVectorSize"] = "16", ["NextValueOp1"] = "TestLibrary.Generator.GetSingle()", ["NextValueOp2"] = "TestLibrary.Generator.GetSingle()", ["ValidateFirstResult"] = "BitConverter.SingleToInt32Bits(result[0]) != ((float.IsNaN(left[0]) || float.IsNaN(right[0])) ? -1 : 0)",                            ["ValidateRemainingResults"] = "BitConverter.SingleToInt32Bits(left[i]) != BitConverter.SingleToInt32Bits(result
```

Comment:
How are the `random.Next(int, int)` values being handled?

Reply:
For now, I have decided to replace only `NextDouble()` in Fma tests to address issue #18467. 

I did more work in my another private branch (https://github.com/echesakovMSFT/coreclr/tree/RemoveTrueRandomnessHardwareIntrinsicsTests) so you can take look. 

In couple words what I did there was
1. Replacing `random.Next() with corrersponding function in CoreCLRTestLibrary
2. Added function to CoreCLRTestLibrary to handle case you pointing to `random.Next(minValueIncl, maxValueExcl)`
3. Modified templates and removed `new Random()` expressions
4. Regenerated all the test files

It seems that we are missing some templates https://github.com/dotnet/coreclr/issues/18467#issuecomment-403175666 so this needs to be fixed before we can fully rely on CoreCLRTestLibrary instead of Random()

Reply:
@tannergooding Does this approach sound right?

Reply:
The overall approach sounds correct.

There is a general work item to move all tests to be templated, but it hasn't been completed yet, but I don't believe those tests are prone to the same flakiness as the FMA tests.

For the current PR, it looks like you modified the `GenerateTests.csx` and `*.template` files, but manually updated the FMA tests rather than regenerate all tests, correct? -- I ask because the tests that are still using `random.Next(int, int)` should fail to compile, otherwise (and I don't see those tests being updated here either).

Reply:
Yes, you are correct. I probably should have not updated `*.template` files in this PR. If I update only `GenerateTests.csx` and FMA tests now and later open another PR addressing integer random values (and include changes to the templates there) would it work?

Reply:
It should. Noting that, for FMA tests, `Random random = new Random()` will still exist, but it will be unused until the the templates and the CoreCLRTestLibrary can be updated to handle all integer types.